packages/alert-dialog/tests/alert-dialog.test.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from "react"
+
+import {
+  render,
+  testA11y,
+  screen,
+  waitFor,
+  fireEvent,
+} from "@chakra-ui/test-utils"
+import {
+  AlertDialog,
+  AlertDialogBody,
+  AlertDialogContent,
+  AlertDialogHeader,
+  AlertDialogOverlay,
+  AlertDialogFooter,
+} from "../src"
+
+interface BasicUsageProps {
+  isOpen?: boolean
+}
+
+const BasicUsage = (props: BasicUsageProps) => {
+  const [isOpen, setIsOpen] = React.useState(props.isOpen || false)
+
+  const onOpen = () => setIsOpen(true)
+  const onClose = () => setIsOpen(false)
+
+  const cancelRef = React.useRef<HTMLButtonElement>(null)
+
+  return (
+    <>
+      <button type="button" onClick={onOpen}>
+        Delete something
+      </button>
+      <AlertDialog
+        isOpen={isOpen}
+        leastDestructiveRef={cancelRef}
+        onClose={onClose}
+      >
+        <AlertDialogOverlay>
+          <AlertDialogContent>
+            <AlertDialogHeader>Please Confirm!</AlertDialogHeader>
+            <AlertDialogBody>
+              Are you sure you want to delete something? This action is
+              permanent, and we're totally not just flipping a field called
+              "deleted" to "true" in our database, we're actually deleting
+              something.
+            </AlertDialogBody>
+            <AlertDialogFooter>
+              <button type="button" ref={cancelRef} onClick={onClose}>
+                Nevermind
+              </button>
+              <button type="button">Yes, delete</button>
+            </AlertDialogFooter>
+          </AlertDialogContent>
+        </AlertDialogOverlay>
+      </AlertDialog>
+    </>
+  )
+}
+
+describe("<AlertDialog />", () => {
```

Comment:
imo we should [avoid nesting when we're testing](https://kentcdodds.com/blog/avoid-nesting-when-youre-testing). It doesn't really add anything to wrap every test in a file in the same block

Reply:
Or at the very least, add it in a separate PR. Since it indents the entire file, it's impossible to tell what's actually changing in some of these files 🙃

Reply:
Generally I agree with everything in Kents article there, obviously, his example is far over the top though and there to make a point.

> The describe function is intended to group related tests together and can provide for a nice way to visually separate different tests, especially when the test file gets big. But I don't like it when the test file gets big. So instead of grouping tests by describe blocks, I group them by file. So if there's a logical grouping of different tests for the same "unit" of code, I'll separate them by putting them in completely different files.

So his point is to avoid `describe` blocks if your test is so large that it should be a separate test file on itself. I don't think that applies here _yet_.

Also I'm somewhat sure if a test fails, we don't have to search for the test name but can use the describe tag right above because it adds indentation on CI too.

Since we can reintroduce it at any point anyways, I'll remove it again later today.

Reply:
on a sidenote, having a `top level describe` [is considered `best practice` by jest itself](https://github.com/jest-community/eslint-plugin-jest/blob/master/src/rules/require-top-level-describe.ts#L8)